envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -210,18 +211,61 @@ class EnvoyClustersFactory(
         return clustersForGroup
     }
 
-    private fun createClusterForGroup(dependencySettings: DependencySettings, cluster: Cluster?): Cluster? {
+    private fun createClusterForGroup(
+        dependencySettings: DependencySettings,
+        cluster: Cluster?,
+        defaultDependencySettings: DependencySettings? = null
+    ): Cluster? {
         return cluster?.let {
             val idleTimeoutPolicy =
-                dependencySettings.timeoutPolicy.connectionIdleTimeout ?: cluster.commonHttpProtocolOptions.idleTimeout
+                dependencySettings.timeoutPolicy.connectionIdleTimeout
+                    ?: defaultDependencySettings?.timeoutPolicy?.connectionIdleTimeout
+                    ?: cluster.commonHttpProtocolOptions.idleTimeout
             Cluster.newBuilder(cluster)
+                .setCircuitBreakers(createCircuitBreakers(dependencySettings))
                 .setCommonHttpProtocolOptions(
                     HttpProtocolOptions.newBuilder().setIdleTimeout(idleTimeoutPolicy)
                 ).build()
         }
     }
 
-    private fun shouldAddDynamicForwardProxyCluster(group: Group) =
+    private fun createCircuitBreakers(
+        dependencySettings: DependencySettings,
+        defaultDependencySettings: DependencySettings? = null
```

Comment:
this argument seems never to be passed to this method or am I wrong

Reply:
here it is passed https://github.com/allegro/envoy-control/pull/302/files/ac7e60c5c2587272caeeceffc3eca2ff979af705#diff-3365f972fd7d855f3310c1d378f6ffc16512b9b659e0ea5a6ba8d985ed20743cR200